src/components/ProductFilter.js

```diff
@@ -0,0 +1,278 @@
+              type="checkbox"
+              onChange={() => dispatch({ type: "sort", payload: null })}
+              checked={state.sort === null}
+            />
+          </label>
+        </div>
+      </div>
+
+      <div className="productFilter_filter">
+        <h4 className="productFilter_filterHeading">Filter By Brand</h4>
+        <div className="productListing_options">
+          {brands.map((entry, index) => {
+            return (
+              <label key={index}>
+                {entry}
+                <input
+                  type="checkbox"
+                  onChange={() => dispatch({ type: "brand", payload: entry })}
+                  checked={state.brand.includes(entry)}
+                />
+              </label>
+            );
+          })}{" "}
+          <label>
+            None
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "brand", payload: "None" })}
+              checked={state.brand.includes("None")}
+            />
+          </label>
+        </div>
+      </div>
+      <div className="productFilter_filter">
+        <h4 className="productFilter_filterHeading">Filter By Rating</h4>
+        <div className="productListing_options">
+          {[1, 2, 3, 4].map((entry, index) => {
+            return (
+              <label key={index}>
+                {entry}+ Star
+                <input
+                  type="checkbox"
+                  onChange={() => dispatch({ type: "rating", payload: entry })}
+                  checked={state.rating === entry}
+                />
+              </label>
+            );
+          })}{" "}
+          <label>
+            None
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "rating", payload: 0 })}
+              checked={state.brand === 0}
+            />
+          </label>
+        </div>
+      </div>
+
+      <div className="productFilter_filter">
+        <h4 className="productFilter_filterHeading">Filter Further</h4>
+        <div className="productListing_options">
+          <label>
+            Only Fast Delivery
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "deliveryToggle" })}
+              checked={state.onlyFastDelivery}
+            />
+          </label>
+          <label>
+            In Stock
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "stockToggle" })}
+              checked={state.onlyInStock}
+            />
+          </label>
+          <label>
+            None
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "resetstockndelivery" })}
+              checked={!state.onlyFastDelivery && !state.onlyInStock}
+            />
+          </label>
+        </div>
+      </div>
+
+      {/* helo */}
+    </div>
+  );
+};
+
+const ProductFilterResponsive = (Component) => {
```

Comment:
You can move this out to a separate file

Reply:
will do that